src/instructlab/sdg/utils/chunkers.py

```diff
@@ -0,0 +1,563 @@
+        return doc_dict
+
+
+class TextSplitChunker(ChunkerBase):
+    def __init__(
+        self,
+        document_contents: List | str,
+        server_ctx_size: int,
+        chunk_word_count: int,
+        output_dir: Path,
+    ):
+        self.document_contents = document_contents
+        self.server_ctx_size = server_ctx_size
+        self.chunk_word_count = chunk_word_count
+        self.output_dir = output_dir
+
+    def chunk_documents(self) -> Dataset:
+        """Naively chunk markdown documents based on the word count provided by the user.
+        Returns:
+            List[str]: List of chunked documents.
+        """
+        num_tokens_per_doc = self._num_tokens_from_words(self.chunk_word_count)
+        if num_tokens_per_doc > int(self.server_ctx_size - 1024):
+            raise ValueError(
+                "Error: {}".format(
+                    str(
+                        f"Given word count ({self.chunk_word_count}) per doc will exceed the server context window size ({self.server_ctx_size})"
+                    )
+                )
+            )
+        if self.document_contents == []:
+            return []
+
+        # Placeholder for params
+        content = []
+        chunk_size = self._num_chars_from_tokens(num_tokens_per_doc)
+        chunk_overlap = _DEFAULT_CHUNK_OVERLAP
+
+        # Using Markdown as default, document-specific chunking will be implemented in separate pr.
+        md_text_splitter = RecursiveCharacterTextSplitter.from_language(
+            language=Language.MARKDOWN,
+            chunk_size=chunk_size,
+            chunk_overlap=chunk_overlap,
+        )
+
+        # Determine file type for heuristics, default with markdown
+        for doc in self.document_contents:
+            # Use regex to remove unnecessary dashes in front of pipe characters in a markdown table.
+            doc = re.sub(r"-{2,}\|", "-|", doc)
+            # Remove unnecessary spaces in front of pipe characters in a markdown table.
+            doc = re.sub(r"\  +\|", " |", doc)
+            temp = md_text_splitter.create_documents([doc])
+            content.extend([item.page_content for item in temp])
+
+        return content
+
+    @staticmethod
+    def _num_tokens_from_words(num_words) -> int:
+        return int(num_words * 1.3)  # 1 word ~ 1.3 token
+
+    @staticmethod
+    def _num_chars_from_tokens(num_tokens) -> int:
+        return int(num_tokens * 4)  # 1 token ~ 4 English character
+
+
+class ContextAwareChunker(ChunkerBase):
+    def __init__(
+        self,
+        document_paths,
+        filepaths,
+        leaf_node_path,
+        output_dir: Path,
+        tokenizer_model_name=None,
+    ):
+        self.document_paths = document_paths
+        self.filepaths = filepaths
+        self.leaf_node_path = leaf_node_path
+        self.output_dir = self._path_validator(output_dir)
+        if tokenizer_model_name is None:
+            self.tokenizer_model_name = "mistralai/Mixtral-8x7B-Instruct-v0.1"
```

Comment:
Why not just make this the default for tokenizer_model_name instead of having it default to None and then override that here?

Reply:
Thanks @jwm4, addressed!

Reply:
@aakankshaduggal is the issue that @khaledsulayman raised going to be a problem with the new implementation?  It does look like if something passes an explicit None that this will fail when it tries to initialize the autotokenizer.

Reply:
yeah, makes sense. I will do a none check to ensure it isn't empty because initializing then